lib/disposable.dart

```diff
@@ -143,10 +145,13 @@ typedef Future<dynamic> Disposer();
 /// without explicit reference to [Disposable]. To do this, we use
 /// composition to include the [Disposable] machinery without changing
 /// the public interface of our class or polluting its lifecycle.
+///
+/// Implementing the [DisposableManager] interface is not required.
 class Disposable implements _Disposable, DisposableManager {
   Completer<Null> _didDispose = new Completer<Null>();
   bool _isDisposing = false;
   List<_Disposable> _internalDisposables = [];
+  Set<Future> _blockingFutures = new Set<Future>();
```

Comment:
Will strong mode complain if `Future` is missing a type, ie this type signature is `Set<Future<dynamic>>`? Would it better to keep a set of null valued futures instead and wrap all pending calls in async wrapper function that gurantees it meets this type instead?

```dart
Set<Future<Null>> _blockingFutures = new Set<Future<Null>>();

/// Other good codes...

Future<T> waitBeforeDispose<T>(Future<T> future) {
     _throwOnInvalidCall(future, 'waitBeforeDispose');
     if (!_blockingFutures.contains(future)) {
       Future<Null> removeFuture;
       Future<Null> opFuture = new Future<Null>(() async {
           await future;
        });
       removeFuture = opFuture.then((T _) {
         _blockingFutures.remove(opFuture);
         _blockingFutures.remove(removeFuture);
      });

      _blockingFutures.addAll([opFuture, removeFuture]);
      return future;
}
```

This is more for explicit type safety which may be worth the indirection for a library that functions with a high degree of type flexibility.